OpenSim/Simulation/Model/CMCActuatorSubsystem.cpp

```diff
@@ -146,14 +142,21 @@ void CMCActuatorSubsystemRep::setSpeedTrajectories(FunctionSet *aSet) {
         _qSet->evaluate(_uWork,1,t);
     }
 
-    for(i=0;i<nq;i++) q[i] = _qWork[i] + _qCorrections[i];
-    for(i=0;i<nu;i++) u[i] = _uWork[i] + _uCorrections[i];
+    /* Hack to obtain a modifiable state in a const method */
+    State& modCompState = const_cast<SimTK::State&>(_completeState);
 
+    // Update the coordinate values to pose the model while computing muscle
+    // controls
+    const CoordinateSet& coords = _model->getCoordinateSet();
+    for (int i = 0; i < nq; ++i) {
+        coords[i].setValue(modCompState, _qWork[i] + _qCorrections[i], i==(nq-1));
```

Comment:
Since we are setting all the coordinate values, I wait to the last coordinate before enforcing the constraints. The default is to enforce constraints on every `setValue()`.  I will add a clarifying comment.